src/mgmt/srv_system.c

```diff
@@ -734,10 +771,22 @@ ds_mgmt_join_handler(struct mgmt_join_in *in, struct mgmt_join_out *out)
 
 	D_DEBUG(DB_TRACE, "rank %u joined in map version %u\n", rank,
 		map_version);
+
 	svc->ms_map_version = map_version;
 	if (in->ji_rank != -1)
 		svc->ms_rank_next = rank_next;
+
+	entry.se_rank = rank;
+	entry.se_uri = in->ji_server.sr_uri;
+	rc = ds_mgmt_group_update(CRT_GROUP_MOD_OP_ADD, &entry,
+				  1 /* nservers */, map_version);
+	if (rc != 0) {
+		rdb_resign(svc->ms_rsvc.s_db, svc->ms_rsvc.s_term);
+		rc = 0;
```

Comment:
Not sure I follow here, so even group update fail, the client will not be notified?

Reply:
If the group update fails, because the system map has already been updated successfully, we consider the Join RPC a success, and reply 0. But in this case this MS leader's local group is stale, so we ask it to step down to avoid handling more MS RPCs.